### PR TITLE
#7563 feat: updates storybook 6.0.28

### PR DESCRIPTION
**Pull request by chriddell**
See wellcometrust/corporate/issues/7563

- updates `storybook` and `@storybook/` packages to 6.0.28 (latest stable)
- adds `html-webpack-plugin@5.0.0-alpha.9` to dev dependencies (this is a workaround for https://github.com/jantimon/html-webpack-plugin/issues/1377 & https://github.com/storybookjs/storybook/issues/9216)
- updates storybook config implementation to match best practice from Storybook 6.0.x docs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "react-html-parser": "^2.0.2",
     "react-paginate": "^6.5.0",
     "shortid": "^2.2.15",
+    "storybook": "^6.0.28",
```
Comment (creido-welly) [inline package.json]: should this be under `devDependencies`?

**Review by creido-welly — APPROVED** [review]
All good, just that one query over dependency vs devDependency

File: package.json (hunk 1 above)
Comment (chriddell) [reply package.json]: Good q, though I think we deploy to https://ghp.wellcomecollection.org/corporate-components/ - so does that mean it should stay in depdendencies?

Comment (chriddell) [reply package.json]: ...then again, that URL is not really production, is it? It's just a deployed version of a testing environment. I think I'll move storybook to devDependencies.

Comment (creido-welly) [reply package.json]: Absolutely. The storybook dependency would become part of the npm package which we definitely don't want

**Review by creido-welly — APPROVED** [review]

**Pull request merged**